[MLIR][LLVMIR] Fix LLVM IR import of ZeroInitializers to constant zero

Bhuvan1527 opened the pull request:
Constant zero aggregate structures are imported to from llvm IR as undef.
This includes for example LandingPad Instructions which have zero value filters, structs. 

This fixes the import to use the zeroOp to materialize a zero-initialized constant.

Bhuvan1527 committed:
[MLIR][LLVMIR] Issue when importing LLVM modules which contain LandingPad clauses

llvm IR with landingPad instructions are not the same when exporting back from llvm mlir dialect to llvm IR.

gysit reviewed (commented):
Thanks for looking into this!

I believe the ConstantAggregateZero codepath can be completely removed from the if after the newly added code now.

Bhuvan1527 committed:
[MLIR][LLVMIR] Issue when importing LLVM modules which contain LandingPad Instructions

Issue:
The problem is how constant zero aggregate structures are imported to llvm.mlir from llvmIR.

gysit reviewed (approved):
LGTM from my end modulo last nits.

Bhuvan1527 commented:
> @Bhuvan1527 do you need someone to land this pr for you?

Yes please. I don't have access to merge it. 
Thanks
